code/modules/admin/sql_notes.dm

```diff
@@ -0,0 +1,186 @@
+	var/target_sql_ckey = sanitizeSQL(target_ckey)
+	if(!notetext)
+		notetext = input(usr,"Write your Note","Add Note") as message
+		if(!notetext)
+			return
+	notetext = sanitizeSQL(notetext)
+	if(!timestamp)
+		timestamp = SQLtime()
+	if(!adminckey)
+		adminckey = usr.ckey
+		if(!adminckey)
+			return
+	var/admin_sql_ckey = sanitizeSQL(adminckey)
+	if(!server)
+		if (config && config.server_name)
+			server = config.server_name
+	server = sanitizeSQL(server)
+	var/DBQuery/query_noteadd = dbcon.NewQuery("INSERT INTO erro_messages (type, targetckey, adminckey, text, timestamp, server) VALUES ('note', '[target_sql_ckey]', '[admin_sql_ckey]', '[notetext]', '[timestamp]', '[server]')")
+	if(!query_noteadd.Execute())
+		var/err = query_noteadd.ErrorMsg()
+		log_game("SQL ERROR adding new note to table. Error : \[[err]\]\n")
+		return
+	if(logged)
+		log_admin("[key_name(usr)] has added a note to [target_ckey]: [notetext]")
+		message_admins("[key_name_admin(usr)] has added a note to [target_ckey]:<br>[notetext]")
+		send2adminirc("[get_key(usr)] has added a note to [target_ckey]: \"[notetext]\"")
+		show_note(target_ckey)
+
+/proc/remove_note(note_id)
+	var/ckey
+	var/notetext
+	var/adminckey
+
+	if(!check_rights(R_INVESTIGATE, 1, usr)) return
+
+	if(!dbcon.IsConnected())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	if(!note_id)
+		return
+	note_id = text2num(note_id)
+	var/DBQuery/query_find_note_del = dbcon.NewQuery("SELECT targetckey, text, adminckey FROM erro_messages WHERE id = [note_id] and type = 'note'")
+	if(!query_find_note_del.Execute())
+		var/err = query_find_note_del.ErrorMsg()
+		log_game("SQL ERROR obtaining targetckey, text, adminckey from erro_messages table. Error : \[[err]\]\n")
+		return
+	if(query_find_note_del.NextRow())
+		ckey = query_find_note_del.item[1]
+		notetext = query_find_note_del.item[2]
+		adminckey = query_find_note_del.item[3]
+	var/DBQuery/query_del_note = dbcon.NewQuery("DELETE FROM erro_messages WHERE id = [note_id]")
+	if(!query_del_note.Execute())
+		var/err = query_del_note.ErrorMsg()
+		log_game("SQL ERROR removing note from table. Error : \[[err]\]\n")
+		return
+	log_admin("[key_name(usr)] has removed a note made by [adminckey] from [ckey]: [notetext]")
+	message_admins("[key_name_admin(usr)] has removed a note made by [adminckey] from [ckey]:<br>[notetext]")
+	send2adminirc("[get_key(usr)] has removed a note made by [adminckey] from [ckey]: \"[notetext]\"")
+	show_note(ckey)
+
+/proc/edit_note(note_id)
+	if(!check_rights(R_INVESTIGATE, 1, usr)) return
+
+	if(!dbcon.IsConnected())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	if(!note_id)
+		return
+	note_id = text2num(note_id)
+	var/target_ckey
+	var/sql_ckey = sanitizeSQL(usr.ckey)
+	var/DBQuery/query_find_note_edit = dbcon.NewQuery("SELECT targetckey, text, adminckey FROM erro_messages WHERE id = [note_id] and type = 'note'")
```

Comment:
dbcon.NewQuery("SELECT \`targetckey\`, \`text\`, \`adminckey\` FROM \`erro_messages\` WHERE \`id\` = [note_id] and \`type\` = 'note'")